x-pack/plugins/observability_ai_assistant/public/analytics/schema.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RootSchema } from '@kbn/analytics-client';
+import { Message } from '../../common';
+import type { Feedback } from '../components/feedback_buttons';
+
+export const MESSAGE_FEEDBACK = 'message_feedback';
+
+export interface MessageFeedback extends Message {
+  feedback: Feedback;
+}
+
+export interface TelemetryEvent {
+  eventType: typeof MESSAGE_FEEDBACK;
+  schema: RootSchema<MessageFeedback>;
+}
+
+export const MESSAGE_FEEDBACK_SCHEMA: TelemetryEvent = {
+  eventType: MESSAGE_FEEDBACK,
+  schema: {
+    '@timestamp': {
+      type: 'text',
+      _meta: {
+        description: 'The timestamp of the message.',
+      },
+    },
+    feedback: {
+      type: 'text',
+      _meta: {
+        description: 'Whether the user has deemed this response useful or not',
+      },
+    },
+    message: {
+      properties: {
+        content: {
+          type: 'text',
+          _meta: {
+            description: 'The response generated by the LLM.',
+            optional: true,
+          },
+        },
+        name: {
+          type: 'text',
+          _meta: {
+            description: 'The name of the function that was executed.',
+            optional: true,
+          },
+        },
+        role: {
+          type: 'text',
+          _meta: {
+            description: 'The actor that generated the response.',
+          },
+        },
+        data: {
+          type: 'text',
+          _meta: {
+            description: '',
```

Comment:
Actually, what is this field used for? :D

Reply:
This is data that is _not_ for the LLM to analyse. E.g., we might use to store a timeseries that we use to display data, but we don't send it over to the LLM because it is too many data points. `content` is what we send over to the LLM.

Reply:
I stand corrected

Reply:
I think we can exclude this from the feedback call